### PR TITLE
Package uninstall option in windows packaging script

### PR DESCRIPTION
**Pull request by marcosd4h**
This PR adds the `-forceuninstall` option to the manage osqueryd PowerShell script.

This option will use `msiexec` to uninstall osquery if present.

This PR is related to this [bug](https://github.com/fleetdm/fleet/issues/8891).

### CODE REVIEW

File: tools/deployment/windows_packaging/manage-osqueryd.ps1 [hunk 1]
```diff
@@ -35,13 +35,14 @@ function Test-IsAdmin {
...
   
-  Write-Host "Usage: $programName (-install|-uninstall|-start|-stop|-help)" -foregroundcolor Yellow
+  Write-Host "Usage: $programName (-install|-uninstall|-forceuninstall|-start|-stop|-help)" -foregroundcolor Yellow
   Write-Host ""
   Write-Host "  Only one of the following options can be used. Using multiple will result in "
   Write-Host "  options being ignored."
   Write-Host "    -install                  Install the osqueryd service"
   Write-Host "    -startupArgs              Specifies additional arguments for the service (only used with -install)"
   Write-Host "    -uninstall                Uninstall the osqueryd service"
+  Write-Host "    -forceuninstall           It uninstalls the installed osquery package"
```
[the remark this answers is not in the document]
Comment (marcosd4h) [reply tools/deployment/windows_packaging/manage-osqueryd.ps1]: @mike-myers-tob thanks Mike for looking into this! I've just fixed the help message description.

**Comment by mike-myers-tob** [comment]
Can you explain a little more how uninstalling the package is different from uninstalling the daemon, if that's what the script normally/previously did? What is it "forcing"? When would someone want to invoke this as opposed to the old way? Is this just the way it should _always_ be uninstalled (maybe just replacing the old behavior instead of adding this new option)?

**Comment by marcosd4h** [comment]
> Can you explain a little more how uninstalling the package is different from uninstalling the daemon, if that's what the script normally/previously did? What is it "forcing"? When would someone want to invoke this as opposed to the old way? Is this just the way it should _always_ be uninstalled (maybe just replacing the old behavior instead of adding this new option)?

Sure. The main difference here is that the old `-uninstall` option just removes the windows service, without removing the actual osquery MSI package entry from the windows's package management (osquery will still appear on the `add or remove program options` for example). This will cause unexpected behaviors, such as not being able to uninstall legacy Osquery entries such as the issue described [here](https://github.com/fleetdm/fleet/issues/8891). Another side effect of `-uninstall` is that no files will be removed from the file system, so the osquery files will not be actually uninstalled as they will still be present on the file system.

The `-forceuninstall` option will actually perform the MSI package uninstall action, effectively removing files, windows services and the osquery package management entry. In order to perform this, the powershell script is relying on the Windows Installer COM object (see [here](https://learn.microsoft.com/en-us/windows/win32/msi/installer-object)) to enumerate the list of installed MSI packages through the [ProductsEx()](https://learn.microsoft.com/en-us/windows/win32/msi/installer-productsex) method, and then calling `msiexec /x` with the` product-id` of the installed osquery package.

I didn't want to override the -uninstall option with this code just for backward-compatibility purposes (I assumed here that someone might be relying on this old behavior) so I ended up adding a new option to uninstall the installed MSI osquery package.

**Comment by marcosd4h** [comment]
> @marcosd4h I'm wondering if this is the right place to put such flag. The `uninstall` option exists because the same script can `install` the **service**. So the expectation of the script is to handle that and some other things of the existing installation.
> 
> Moreover this script is installed with the MSI, and when you uninstall it, it will get removed; I don't think it's supposed to exist outside of it (and so handle the MSI package).

@Smjert Thanks for your feedback and clarification on the script goal. This is very useful! I agree that this might not be the right place to put this feature. The idea was to allow osquery users to uninstall previous osquery instances through the script. We were thinking of pointing out customers to grab and use this script if they want to do that.

I'm going to close the PR and move the functionality somewhere else as this script is not the right place to distribute the uninstall logic. I'll comment on the approach taken before closing the PR. Thanks!

**Comment by marcosd4h** [comment]
The unistall logic is going to be delivered as part of the Fleet installer. Thanks everyone for the comments made here!

**Pull request closed without merging**